src/triage/component/architect/feature_dictionary.py

```diff
@@ -0,0 +1,35 @@
+from triage.component.architect.utils import remove_schema_from_table_name
+from triage.util.structs import FeatureNameList
+from collections.abc import Iterable
+
+
+class FeatureDictionary(dict):
+    """A feature dictionary, consisting of table names as keys and column names as values
+
+    If a list of feature_blocks is passed, will initialize the feature dictionary with their data.
+    """
+    def __init__(self, feature_blocks=None, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        for feature_block in feature_blocks:
+            cleaned_table = remove_schema_from_table_name(
+                feature_block.final_feature_table_name
+            )
+            self[cleaned_table] = feature_block.feature_columns
+
+    def __setitem__(self, table, feature_names):
+        if not isinstance(table, str):
+            raise ValueError("key of FeatureDictionary objects represents a table "
```

Comment:
`TypeError` was born to be a star, and I _think_ this is its 15 minutes?

```suggestion
            raise TypeError("key of FeatureDictionary objects represents a table "
```